### PR TITLE
Update UMAT interface vector order to typical use

### LINKED ISSUE #20111: UMAT interface stress/strain vector order
## Bug Description
The strain/stress vector order convention is different in MOOSE compared to commercial software. This can be an annoyance to users and lead to error. This was brought up by @ngrilli:

https://github.com/idaholab/moose/discussions/20109#discussion-3826430

## Steps to Reproduce
Run a multiple-element problem with shear loading.

## Impact
The current order convention in the UMAT interface in MOOSE requires users to adapt their external routines. This seems unnecessary and error-prone.

### PR DESCRIPTION
**Pull request by recuero**
Closes #20111

### CODE REVIEW

**Comment by dschwen** [comment]
How is this not diffing more things?!

**Comment by recuero** [comment]
Because we don't have real life tests!!!

**Comment by dschwen** [comment]
Let's make sure to change that in this pr then. Will require an anisotropic elasticity tensor

File: modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C [hunk 1]
```diff
@@ -170,8 +170,10 @@ AbaqusUMATStress::computeQpStress()
 
   // Pass through updated stress, total strain, and strain increment arrays
   static const std::array<Real, 6> strain_factor{{1, 1, 1, 2, 2, 2}};
+  // Account for difference in vector order convention: yz, xz, xy (MOOSE)  vs xy, xz, yz
+  // (commercial software)
   static const std::array<std::pair<unsigned int, unsigned int>, 6> component{
-      {{0, 0}, {1, 1}, {2, 2}, {1, 2}, {0, 2}, {0, 1}}};
+      {{0, 0}, {1, 1}, {2, 2}, {0, 1}, {0, 2}, {1, 2}}};
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C:176]: By "typical use" you mean not standard Voigt notation order?

Comment (recuero) [reply modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C:176]: I was trying to refer to the use of the UMAT interface. I guess there can be a variety of not standard Voigt order.

**Comment by recuero** [comment]
> Let's make sure to change that in this pr then. Will require an anisotropic elasticity tensor

For the models that we are testing, it may require an anisotropic elasticity tensor. But we would still be cooking up a problem for this particular issue (I was not referring to this, but to the lack of more advanced UMAT routines). Do you want to have a test with some elastic anisotropy added to this PR?

File: modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C [hunk 2]
```diff
@@ -280,34 +282,32 @@ AbaqusUMATStress::computeQpStress()
...
-      _aqDDSDDE[17], // C3312
-      _aqDDSDDE[21], // C2323
-      _aqDDSDDE[22], // C2313
-      _aqDDSDDE[23], // C2312
-      _aqDDSDDE[28], // C1313
-      _aqDDSDDE[29], // C1312
-      _aqDDSDDE[35]  // C1212
-  }});
+  // Build Jacobian matrix from UMAT's Voigt non-standard order to fourth order tensor.
+  _jacobian_mult[_qp].zero();
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C:294]: remove
```suggestion
```

Comment (recuero) [reply modules/tensor_mechanics/src/materials/abaqus/AbaqusUMATStress.C]: Just did.

**Review by dschwen — APPROVED** [review]
Looks good to me

**Pull request merged**